modules/data/src/main/java/io/fluo/webindex/data/LoadHdfs.java

```diff
@@ -71,37 +72,38 @@ public static void main(String[] args) throws Exception {
     log.info("Loading {} files into Fluo from {}", loadPaths.size(), dataDir);
 
     SparkConf sparkConf = new SparkConf().setAppName("webindex-load-hdfs");
-    JavaSparkContext ctx = new JavaSparkContext(sparkConf);
+    try (JavaSparkContext ctx = new JavaSparkContext(sparkConf)) {
 
-    JavaRDD<String> paths = ctx.parallelize(loadPaths, loadPaths.size());
+      JavaRDD<String> paths = ctx.parallelize(loadPaths, loadPaths.size());
 
-    paths.foreachPartition(iter -> {
-      final FluoConfiguration fluoConfig = new FluoConfiguration(new File("fluo.properties"));
-      FileSystem fs = IndexEnv.getHDFS(hadoopConfDir);
-      try (FluoClient client = FluoFactory.newClient(fluoConfig);
-          LoaderExecutor le = client.newLoaderExecutor()) {
-        iter.forEachRemaining(path -> {
-          Path filePath = new Path(path);
-          try {
-            if (fs.exists(filePath)) {
-              FSDataInputStream fsin = fs.open(filePath);
-              ArchiveReader reader = WARCReaderFactory.get(filePath.getName(), fsin, true);
-              for (ArchiveRecord record : reader) {
-                Page page = ArchiveUtil.buildPageIgnoreErrors(record);
-                if (page.getOutboundLinks().size() > 0) {
-                  log.info("Loading page {} with {} links", page.getUrl(), page.getOutboundLinks()
-                      .size());
-                  le.execute(PageLoader.updatePage(page));
+      paths.foreachPartition(iter -> {
+        final FluoConfiguration fluoConfig = new FluoConfiguration(new File("fluo.properties"));
+        FileSystem fs = IndexEnv.getHDFS(hadoopConfDir);
+        try (FluoClient client = FluoFactory.newClient(fluoConfig);
+            LoaderExecutor le = client.newLoaderExecutor()) {
+          iter.forEachRemaining(path -> {
+            Path filePath = new Path(path);
+            try {
+              if (fs.exists(filePath)) {
+                FSDataInputStream fsin = fs.open(filePath);
+                ArchiveReader reader = WARCReaderFactory.get(filePath.getName(), fsin, true);
+                for (ArchiveRecord record : reader) {
+                  Page page = ArchiveUtil.buildPageIgnoreErrors(record);
+                  if (page.getOutboundLinks().size() > 0) {
+                    log.info("Loading page {} with {} links", page.getUrl(), page
+                        .getOutboundLinks().size());
+                    le.execute(PageLoader.updatePage(page));
+                  }
                 }
               }
+            } catch (IOException e) {
+              log.error("Exception while processing {}", path, e);
             }
-          } catch (IOException e) {
-            log.error("Exception while processing {}", path, e);
-          }
-        });
-      }
-    });
+          });
+        }
+      });
 
-    ctx.stop();
+      ctx.stop();
```

Comment:
Not sure this is needed since close is called by try w/ resources.  Close seems to call stop, but not sure.

https://github.com/apache/spark/blob/f9ae99fee13681e436fde9899b6a189746348ba1/core/src/main/scala/org/apache/spark/api/java/JavaSparkContext.scala#L637